### DISCUSSION TITLE
edge functions how deploy _shared folder

### DISCUSSION THREAD

**Discussion by Gi972** [supabase/supabase#8723] [Questions] [answered]
Hello everybody,

In the guide , section:  Edge function => Organizing your Edge Functions

It is recommend developing “fat functions”. 
In local, I use the same organisation an I created folder "_shared" In and it's work.
But how I can deploy this folder on supabase?

**Comment by GaryAustin1** [comment, marked as answer]
When you deploy `supabase functions deploy hello-world` the _shared folder is under functions folder (where the function code resides) and according to the docs any folder starting underscore will get pulled into the function bundle by deploy.

**Reply by AyoPrez** [reply]
I don't get it. If I have a bunch of common code inside the “_shared” folder, how can deploy all that code?
When I try to deploy any folder with a name starting by an underscore, I get the error message:
` Invalid Function name. Must start with at least one letter, and only include alphanumeric characters, underscores, and hyphens. (^[A-Za-z][A-Za-z0-9_-]*$)`

**Reply by GaryAustin1** [reply]
I believe that _ folders get pulled in with any function in the same parent folder (functions in the examples).

**Reply by AyoPrez** [reply]
In that case, I should be doing something wrong, because I write in terminal `supabase functions deploy _test --project-ref <ref>` and I get the error message `Invalid Function name. Must start with at least one letter, and only include alphanumeric characters, underscores, and hyphens. (^[A-Za-z][A-Za-z0-9_-]*$)`.
Is there another way to upload files to the functions? Can you tell me what I'm doing wrong?

**Reply by GaryAustin1** [reply]
You don't load _test.  You load a function also in the same directory and _test will also get loaded with it.
functions
   
```
_test
myfunctiondir
         index.ts
 ```         
You deploy myfunctiondir and _test files will also get loaded.

**Reply by AyoPrez** [reply]
Oh, ok. In that case, the documentation is a bit confusing in that point. As @Gi972 mentioned in the original post, the documentation (https://supabase.com/docs/guides/functions/best-practices) and the examples in Github (https://github.com/supabase/supabase/tree/master/examples/edge-functions/supabase/functions) make understand (or at least I understand it that way) that inside "_shared" we can add files without the need of create functions, and that those files will be available to be used by the other functions created.
